test/e2e/test_worker_status.py

```diff
@@ -130,3 +103,92 @@ def check_worker_processes_exist() -> None:
 
         # Check that there are worker processes running
         check_worker_processes_exist()
+
+    @pytest.mark.skipif(
+        os.environ["OPERATING_SYSTEM"] == "linux",
+        reason="Windows specific test",
+    )
+    def test_windows_worker_restarts_process(
+        self,
+        deadline_resources,
+        deadline_client: DeadlineClient,
+        class_worker: EC2InstanceWorker,
+    ) -> None:
+        # Verifies that Windows Worker service restarts the process when we start/stop worker process
+
+        assert class_worker.worker_id is not None  # This fixes linter type mismatch
+
+        assert is_worker_started(
+            deadline_client=deadline_client,
+            farm_id=deadline_resources.farm.id,
+            fleet_id=deadline_resources.fleet.id,
+            worker_id=class_worker.worker_id,
+        )
+
+        # First check that the worker service is running
+
+        @backoff.on_exception(
+            backoff.constant,
+            Exception,
+            max_time=30,
+            interval=2,
+        )
+        def check_service_is_running() -> None:
+            # The service should be running
+            service_check_result = class_worker.send_command(
+                '(Get-Service -Name "DeadlineWorker").Status'
+            )
+            assert (
+                service_check_result.exit_code == 0
+            ), "Unable to check whether DeadlineWorker service is running"
+            assert (
+                "Running" in service_check_result.stdout
+            ), f"DeadlineWorker service is in unexpected status {service_check_result.stdout}"
+
+        check_service_is_running()
+
+        # Check that the worker process is running
+
+        def check_worker_processes_exist() -> None:
+            process_check_result = class_worker.send_command("Get-Process pythonservice")
+
+            assert process_check_result.exit_code == 0, "Worker agent process is not running"
```

Comment:
Should the quote for the error start with an `f`?

Reply:
Nope, f indicates f string which can be populated with variables. We do not need a variable inside of the string in this case.